### ISSUE TITLE
Support "multi-choice" input type for manual workflows

### ISSUE THREAD

**Issue by DpoBoceka** [actions/runner#2076]
**Describe the enhancement**
Current set of input types doesn't cover some use-cases for manual workflows

We use a single gh-action as an entrypoint of deploying packages from our monorepo, and sometimes we need to manually deploy several packages simultaneously. Up until now we  just went with a set of Booleans, so we could tick all the packages we needed to run our CI/CD against via UI. However, we speedily reached a [limit of 10 input values](https://github.com/actions/runner/issues/1928)

Introducing an input type that would store a, say,  json array could help:
```yaml
on:
  workflow_dispatch:
    inputs:
      name:
        type: multi-choice
        description: select packages
        options: 
        - go-service1
        - go-service2
        - py-service1
```

...and in fact would be more logical with less copy-paste overhead for our use-case

**Comment by Ido-Ozeri** [comment]
+1

**Comment by BondLi1** [comment]
+1

**Comment by paul-gladoon** [comment]
+1

**Comment by kaira** [comment]
+1

**Comment by mato1411** [comment]
+1

**Comment by MatanHeledPort** [comment]
+1

**Comment by danielsinai** [comment]
+1

**Comment by kc-mzb** [comment]
+1

**Comment by Dimitris-Mil** [comment]
+1

**Comment by victorchurikovcommonground** [comment]
+1

**Comment by priyanthabuddhika** [comment]
I would be really great feature to have since many use cases are there.

**Comment by joshua-reddish** [comment]
+1

**Comment by vincda** [comment]
+1

**Comment by patriziobrunops** [comment]
+1

**Comment by japseabery** [comment]
+1

**Comment by chillybug** [comment]
+1
I want to use this feature to pass dynamic matrix strategy for manual workflows.

**Comment by Rockdrago12** [comment]
rodrigomancillamabr@gmail.com
+1

**Comment by paulspiegel** [comment]
+1

**Comment by jwilson232** [comment]
+1

**Comment by dcruzrinkel** [comment]
+1

**Comment by vbotay** [comment]
+1

**Comment by pearlberry** [comment]
+1

**Comment by djoshi0** [comment]
+1

**Comment by gheorghesava-mck** [comment]
+1

**Comment by galz-cyera** [comment]
+1

**Comment by n-tangufor** [comment]
+1

**Comment by NP-compete** [comment]
+1

**Comment by rgildein** [comment]
+1

**Comment by seos-vcbs** [comment]
+1

**Comment by rochanaatapattu** [comment]
+1

**Comment by hkpanchani** [comment]
+1

**Comment by sergio-toro** [comment]
+1

**Comment by fouringiz** [comment]
+1

**Comment by fduch2k** [comment]
+1

**Comment by shaul75** [comment]
+1

**Comment by ClementLucasPopchef** [comment]
+1

**Comment by JacopKane** [comment]
This is a highly needed feature because of the 10 input limit. The only alternative way is sending options by commas which is very error-prone.

**Comment by mehicdado** [comment]
+1 We have the same issue. It would be great to have this as a possibility so multiple options can be selected simultaneously. For example, one could choose which services, and images, to be built and/or deploy.

**Comment by ygthns** [comment]
+1

**Comment by yorksonjeen** [comment]
+1

**Comment by hopeai** [comment]
+1

**Comment by karanadep-dexcom** [comment]
+1

**Comment by marcelodevops** [comment]
+1

**Comment by kaye-alvarado** [comment]
+1 even with limits (max 10) is very helpful

**Comment by pritom007** [comment]
+1

**Comment by DamianPilarczyk** [comment]
+1 would love to use it with categories for E2E tests

**Comment by biandratti** [comment]
+1

**Comment by EvenGal** [comment]
+1

**Comment by izi-p** [comment]
This is badly needed 🔥

**Comment by sdesalas** [comment]
+1

**Comment by alisinabh** [comment]
+1

**Comment by saerosV** [comment]
+1, this would certainly help a lot of different use cases.

**Comment by koroldavid** [comment]
+1

**Comment by pritom007** [comment]
> **Describe the enhancement** Current set of input types doesn't cover some use-cases for manual workflows
> 
> We use a single gh-action as an entrypoint of deploying packages from our monorepo, and sometimes we need to manually deploy several packages simultaneously. Up until now we just went with a set of Booleans, so we could tick all the packages we needed to run our CI/CD against via UI. However, we speedily reached a [limit of 10 input values](https://github.com/actions/runner/issues/1928)
> 
> Introducing an input type that would store a, say, json array could help:
> 
> ```yaml
> on:
>   workflow_dispatch:
>     inputs:
>       name:
>         type: multi-choice
>         description: select packages
>         options: 
>         - go-service1
>         - go-service2
>         - py-service1
> ```
> 
> ...and in fact would be more logical with less copy-paste overhead for our use-case

Also it would be nice to have a option to take user inputs as one of the multi-choice.
```yaml
on:
  workflow_dispatch:
    inputs:
      name:
        type: multi-choice
        description: select packages
        options: 
        - go-service1
        - go-service2
        - py-service1
        - input()
```

**Comment by videap** [comment]
+1

**Comment by chad-bekmezian-snap** [comment]
+1

**Comment by KaramAgha** [comment]
+1 on this

**Comment by pozimkiewicz** [comment]
+1

**Comment by acuD1** [comment]
+1 🙏

**Comment by nidhinnru** [comment]
+1

**Comment by nikola-jokic** [comment]
Hey everyone, this is not really a runner issue :relaxed:. We have communicated this issue to the correct team, but I will leave it open because it gained so much attention :relaxed:

**Comment by KenticoDevTrev** [comment]
Just running into this as well, although mainly on the ability to have a "input()" as someone mentioned.  We want to be able to say "select a release to deploy" where it could be the "latest" or a specific one.

**Comment by brandonfl** [comment]
Also needing this, so +1 from my side :smile:

**Comment by ytariqrc** [comment]
+1

**Comment by aga98** [comment]
+1

**Comment by MrEquableJarl** [comment]
+1

**Comment by agoel2** [comment]
+1

**Comment by MunchyYDL** [comment]
It would be an awesome addition to the choice type. +1

**Comment by yanjiefan** [comment]
+1

**Comment by ori-fantini-uala** [comment]
+1

**Comment by lepskiy** [comment]
+1

**Comment by sumitatdeliverect** [comment]
+1

**Comment by RoshmiB** [comment]
+1

**Comment by aviramshv** [comment]
+1

**Comment by nickkent88** [comment]
+1

**Comment by Iapachonp** [comment]
+1

**Comment by tranducminh** [comment]
+1

**Comment by hozdenoren** [comment]
+1

**Comment by aprietop** [comment]
+1

**Comment by lovelyCARDINAL** [comment]
+1

**Comment by FabrizioCafolla** [comment]
+1

**Comment by AxelTahmid** [comment]
+1

**Comment by suryawanshi-rakesh** [comment]
+1

**Comment by andrzejTheBauer** [comment]
+1

**Comment by thanthtetzin** [comment]
+1

**Comment by breck-ohlson** [comment]
+1

**Comment by ssay-work** [comment]
+1 LETS GO!

**Comment by kamilchodola** [comment]
+1 - plenty of use cases for that.

**Comment by stdevMac** [comment]
+1 Definitely would make a difference

**Comment by shashankrathoretelus** [comment]
A Must Must required Feature.

**Comment by moerafiee** [comment]
+1

**Comment by dmorgun** [comment]
+1 please do this

**Comment by aagresta-igt** [comment]
+1

**Comment by pedroacarrasco** [comment]
👍  must require!!!

**Comment by ahiggins-pella** [comment]
+1

**Comment by NickDiz** [comment]
+1

**Comment by tuananhlai** [comment]
+1 💯

**Comment by niravshah2705** [comment]
+1

**Comment by Mikkelhost** [comment]
+1

**Comment by minhnd410** [comment]
+1

**Comment by nickswiss** [comment]
+1

**Comment by algo7** [comment]
+1

**Comment by uthornbl** [comment]
+1

**Comment by OlekDataTeamVasco** [comment]
+1

**Comment by pritom007** [comment]
It's been quite a while. Are we gonna see any change soon?

**Comment by shaylevin89** [comment]
+1

**Comment by vphanideep** [comment]
+1

**Comment by VikramReddy007** [comment]
+1

**Comment by VikramReddy007** [comment]
Hi Dev team, 

Requesting you to reply in this thread whether this request can be implemented in the near future. Appreciate your effort.

**Comment by zaid0080** [comment]
+1,
Stuck before of this, i belive this would be great feature if it comes out. thanks

**Comment by gusfcarvalho** [comment]
+1 :)

**Comment by mikefelder** [comment]
+1

**Comment by bloonguyen1207** [comment]
+1

**Comment by Loni21** [comment]
+1

**Comment by EstebanBorai** [comment]
+1

**Comment by bryanmacfarlane** [comment]
have you tried multiline inputs?

e.g. in yaml
```yaml
 options: |
     foo
     bar
     baz
```

And then the actions code can split on lines and even validate the allowed values

**Comment by denis-flowwow** [comment]
+1

**Comment by wonderphil** [comment]
+1

**Comment by petrushinvs** [comment]
+1

**Comment by yoku-2093** [comment]
+1

**Comment by bryanmacfarlane** [comment]
If you're going to +1 this request, can you at least talk to why the existing feature set plus built in yaml solutions + toolkit doesn't address the scenario?

It's possible with yaml + toolkit for published actions today.

https://github.com/actions/runner/issues/2076#issuecomment-1777834760

**Comment by GuillaumeMilan** [comment]
@bryanmacfarlane the author @DpoBoceka has already made a great effort describing why it would be nice to add it.
Nothing is not doable at the moment without it. But it is pain to work with a set of boolean if you want to add options regularly.

**Comment by rastitkac** [comment]
+1
This would be very appreciated addition. At least from the folks like me, who would use it for deployments to multiple (selectable) environments/Installations.

**Comment by cardasac** [comment]
+1
Would remove a lot of copy pasting indeed and be very beneficial to quite a few use cases.

**Comment by bryanmacfarlane** [comment]
> But it is pain to work with a set of boolean if you want to add options regularly

@GuillaumeMilan 

The [suggestion](https://github.com/actions/runner/issues/2076#issuecomment-1777834760) did not mention a set of booleans.  It was to use native yaml multiline feature that already exists.  actions js side you can split on lines easily and do whatever validation is needed.

**Comment by P1119r1m** [comment]
+1
This feature will be extremely useful when we need to select multiple options.
For example:
- Build Archs
  - x86
  - arm
  - aarch64

**Comment by bryanmacfarlane** [comment]
@P1119r1m per the suggestion I think you can achieve it today with standard existing yaml features

In the workflow:
```
buildArchs: |
     x86
     arm
     aarach64
```

In the action js split on lines and validate value is one of a defined array.  That also allows the actions to determine if it wants to be case sensitive or not etc. in code.

**Comment by P1119r1m** [comment]
@bryanmacfarlane Thanks for your suggestion!
Does this feature also work with user workflows (when the user selects multiple features to test)?

**Comment by bryanmacfarlane** [comment]
Yes, the buildArchs: above is what the user would put in their workflow.  The action code itself would read and validate the supplied options.

**Comment by bryanmacfarlane** [comment]
I can whip up of a quick example action if that helps ...

**Comment by dzuelke** [comment]
Your proposal sounds like it requires a text box for entry and comparison, @bryanmacfarlane. The problem with that is that
1. the user must know all possible values;
2. typos will happen

What the OP wants is basically a list of checkboxes for when a user manually dispatches a workflow.

This should then be available in an input format that can easily be fed into a matrix.

Example:

<img width="383" alt="Screenshot 2023-11-30 at 02 28 12" src="https://github.com/actions/runner/assets/27900/420d3752-8524-4584-904a-0ee22f679217">

Right now, this requires the following:

```yaml
on:
  workflow_dispatch:
    inputs:
      stack-heroku-20:
        description: 'Sync heroku-20 packages'
        type: boolean
        default: true
        required: false
      stack-heroku-22:
        description: 'Sync heroku-22 packages'
        type: boolean
        default: true
        required: false
      dry-run:
        description: 'Only list package changes, without syncing'
        type: boolean
        default: false
        required: false

jobs:
  stack-list:
    runs-on: ubuntu-22.04
    outputs:
      stacks: ${{ steps.list-stacks.outputs.matrix }}
    steps:
      - name: Install jq tool
        run: |
          sudo apt-get update
          sudo apt-get install jq
      - id: list-stacks
        name: Generate list of stacks to sync based on input checkboxes
        run: |
          set -o pipefail
          stacks=(${{ inputs.stack-heroku-20 == true && 'heroku-20' || ''}} ${{ inputs.stack-heroku-22 == true && 'heroku-22' || ''}})
          echo -n "matrix=" >> "$GITHUB_OUTPUT"
          printf "%s\n" "${stacks[@]}" | jq -jcRn '[inputs|select(length>0)]' >> "$GITHUB_OUTPUT"

  sync:
    needs: [stack-list]
    if: ${{ needs.stack-list.outputs.stacks != '[]' && needs.stack-list.outputs.stacks != '' }}
    runs-on: ubuntu-22.04
    strategy:
      matrix:
        stack: ${{ fromJSON(needs.stack-list.outputs.stacks) }}
```

Drawbacks:
- maximum of ten such input values, which often is not enough
- nasty to generate a matrix value from it (just look at the job...)
- not possible to enforce at the input definition level that at least one item must be picked

Instead it should be something like this:

```yaml
on:
  workflow_dispatch:
    inputs:
      stacks:
        description: 'Sync packages for'
        type: multi-choice
        options: [heroku-20, heroku-22]
        default: [heroku-20, heroku-22]
        required: true
      dry-run:
        description: 'Only list package changes, without syncing'
        type: boolean
        default: false
        required: false

jobs:
  sync:
    runs-on: ubuntu-22.04
    strategy:
      matrix:
        stack: ${[ inputs.stacks ]}
```

**Comment by bryanmacfarlane** [comment]
I see.  The actual request is about the UI options during manual dispatch in the web UI vs the options in the yaml?  That's more about a yaml request to inform the web ui and less about the runner execution (where this issue is logged).   

The request did not make that clear.

If this is a a feature request on the service then it should probably be a feature request on the actions service and not the runner.

**Comment by dzuelke** [comment]
> If this is a a feature request on the service then it should probably be a feature request on the actions service and not the runner. Here is the link for [feature requests for the service](https://github.com/github/feedback/discussions/categories/actions-and-packages-feedback)

That link redirects me to https://github.com/orgs/community/discussions, is that intended, @bryanmacfarlane ?

**Comment by bryanmacfarlane** [comment]
Yeah, it is.  That's where discussions happen for feature requests and product reads those.  You can also upvote those and they will rise to the top.   Here's the [category for Actions](https://github.com/orgs/community/discussions/categories/actions).  BTW, I found [this one](https://github.com/orgs/community/discussions/12029) which is not the same but might be related to this one (e.g. you could dynamically populate list of environments etc).  If I find a better place I'll let you know :)

**Comment by dzuelke** [comment]
FWIW, just for giggles, I tried

```yaml
on:
  workflow_dispatch:
    inputs:
      stack[heroku-20]:
        …
```

But that causes a white page at https://github.com/heroku/heroku-buildpack-php/actions/manual when manually dispatching a workflow 🤠

**Comment by P1119r1m** [comment]
@dzuelke @DpoBoceka 
Also, as I understand from @bryanmacfarlane, we probably need to create a feature request instead of an issue to add:
```
on:
  workflow_dispatch:
    inputs:
      name:
        type: multi-choice
        description: select packages
        options: 
        - go-service1
        - go-service2
        - py-service1
```

It seems that feature requests mentioned in [Issues](https://github.com/actions/runner/issues) are simply being ignored.

**Comment by bryanmacfarlane** [comment]
Small feature requests scoped to the repo can get attention and engineering does look over issues.  However, product (making priority decisions on new features) looks much closer at the product feedback ( discussions) and their upvotes coming from customers.  If someone wants to create the discussion and we get all the folks here who +1 to up vote there it will help :) 

I was just going to see about updating our issue template but I checked and it's already linking to the discussions for feature requests.



![Image](https://github.com/actions/runner/assets/919564/09ea7725-f31e-40f5-8148-fb1604d62282)

**Comment by alxgomz** [comment]
That one looks exactly the same request
I think we should upvote it as mentioned by @bryanmacfarlane https://github.com/orgs/community/discussions/34298

**Comment by connecty-in** [comment]
+1

**Comment by yehonadav-feel** [comment]
+1

**Comment by canack** [comment]
Hi 👋🏻 I've created a terminal tools the manage GitHub Actions more easy.
Some people use json input to get past the maximum input value of 10. It works but It being hard parse json when you need a trigger a new workflow.
 
This tool parse json input automatically and you can trigger your workflow more easy&fast.
GAMA : https://github.com/termkit/gama

Here is example workflow:
```yaml
name: Workflow With JSON

on:
  workflow_dispatch:
    inputs:
      services:
        description: "A JSON field for external services"
        required: true
        default: '{
          "backend-ref": "master",
          "frontend-ref": "master",
          "db-scripts-ref": "master"
          }'
      server:
        description: 'Server'
        type: choice
        required: true
        options:
          - '77'
          - '78'
          - '79'
          - '80'
          - '81'
          - '82'
          - '83'
        default: '80'
      version:
        description: 'Version'
        type: string
        required: true
        default: '1.0.0'
      boolean_flag:
        description: 'Boolean Flag'
        type: boolean
        required: true
        default: true
      number:
        description: 'Number'
        type: number
        required: true
        default: 1
```

The GAMA output:
<img alt="gama" src="https://github.com/community/community/assets/26650249/7874941a-0793-4de4-a104-4441987d9b2a" width="600"/>

**Comment by AlixH** [comment]
Up

**Comment by KenticoDevTrev** [comment]
Anyone else feel like no one is watching these issues and doesn't care? We have over 100 requests on this and it's been open for years

**Comment by byronlian-paypay** [comment]
Up

**Comment by liaddrori1** [comment]
+1

**Comment by alon-apono** [comment]
+1

**Comment by jdegeorgeIC** [comment]
+100

**Comment by thibault-lepeu** [comment]
+1

**Comment by payam-aibees** [comment]
up

**Comment by Joustie** [comment]
+1

**Comment by joaoluiznaufel** [comment]
+1

**Comment by michal-maslaniec-tf** [comment]
+1

**Comment by philipffmc** [comment]
+1

**Comment by zacharyffmc** [comment]
+1

**Comment by yoni-amikam** [comment]
+1

**Comment by apsamuel** [comment]
This would be a game changer for me

**Comment by aj-astra** [comment]
+1

**Comment by Ryang20718** [comment]
+1

**Comment by kris-luminar** [comment]
👍

**Comment by joaoluiznaufel** [comment]
+1

**Comment by TsvetoslavVasev** [comment]
+1

**Comment by 9Rune5** [comment]
They have type: choice now:
https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype

That said, how many found themselves on this page because they wanted a similar input type for other events too?

**Comment by alxgomz** [comment]
> They have type: choice now: https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype
> 
> That said, how many found themselves on this page because they wanted a similar input type for other events too?

`type: choice` renders a dropdown menu which is not suitable for multiple choices, hence this issue

**Comment by alxgomz** [comment]
> They have type: choice now: https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions#onworkflow_dispatchinputsinput_idtype
> 
> That said, how many found themselves on this page because they wanted a similar input type for other events too?

`type: choice` renders a dropdown menu which is not suitable for multiple choices, hence this issue

> That one looks exactly the same request: https://github.com/orgs/community/discussions/34298 I think we should upvote it as mentioned by @bryanmacfarlane

Reminder: feature requests should be upvoted instead of +1 on issues 🙏

**Comment by ThiagoFelippi** [comment]
+1 pleasee

**Comment by amrud** [comment]
+1

**Comment by tsarenkotxt** [comment]
+1

**Comment by ralphotowo** [comment]
+1

**Comment by ihorhorovyi-vitech** [comment]
+1

**Comment by guillemreverse** [comment]
+1

**Comment by Teja-Cuber** [comment]
+1 pretty please 🫰🏻

**Comment by RitikaLaddha** [comment]
+1

**Comment by Adri99x** [comment]
+42069

**Comment by MichaelEureka** [comment]
+1

**Comment by dmkulazhenko** [comment]
+1

**Comment by jan-navratil-IOC** [comment]
+1

**Comment by simantaturja** [comment]
+1

**Comment by aliaksei-smaliuk** [comment]
+1

**Comment by tnfolayan** [comment]
+1

**Comment by rickpasetto** [comment]
+1

**Comment by senser-davidfurman** [comment]
+9999

**Comment by xihajun** [comment]
+1 pleasee

**Comment by liqunn** [comment]
+999999999999999999999999, please support this

**Comment by grmenon** [comment]
& one more +2 ... with a please....

**Comment by richdapice** [comment]
+1

**Comment by yusijs** [comment]
> +1 please
> 
> btw, we can use for a workaround
> 
> ```
> description: 'Enter options separated by commas (e.g., "option1,option2")'
>     required: true
>     type: text
> ```

Problem with this is that in some cases the selection is rather verbose, and not necessarily triggered by developers. For example, we would like to have our QA engineers trigger deployments where they select the apps to deploy from a monorepo. This can be a bit tedious since the selections are very specific, so having them in a multi-select dropdown would be a massive improvement in the workflow.

**Comment by paul-uz** [comment]
2 years since this initial request. Any updates?

This is something we need in order to move away from Jenkins.

**Comment by jdockeray** [comment]
+1

**Comment by rbuchss** [comment]
Please add this! 🥺

**Comment by ehsannm** [comment]
+99

**Comment by NedWilbur** [comment]
+1

**Comment by vmjorge** [comment]
+1

**Comment by AviyaBaumgarten** [comment]
+1

**Comment by vinitshahdeo** [comment]
+1

**Comment by Zero-Fanker** [comment]
+1

**Comment by ENEMBI** [comment]
+1

**Comment by HimanshuD-Groove** [comment]
+1

**Comment by peterrattew** [comment]
I would love to see some more advanced input types with dynamic opinions.

**Comment by Fedoseew** [comment]
+1

**Comment by riga** [comment]
+1

An accompanying function, say,

```js
hasOption(github.event.inputs.<the_option_input>, 'specific_option')
```

that checks if a specific option is present would also be helpful. Certainly better than the current *workaround* of accepting a string containing comma-separated options, manually checking if it contains a certain option and then **handling leading / trailing commas**.

For instance, checking that `py311` is not contained (as an option) in `py311-cuda,py312` is possible of course, but not as simple as it could be given that this is likely a common thing people do.

**Comment by InTheNooB** [comment]
+1

**Comment by pamit** [comment]
+1

**Comment by dromaniuk-gms** [comment]
+1

**Comment by inarendran** [comment]
+1

**Comment by magesh-xseed** [comment]
+1

**Comment by dboriichuk** [comment]
+1

**Comment by pmatthews05** [comment]
+1

**Comment by mdesco-crgl** [comment]
+1

**Comment by zyv4yk** [comment]
+1

**Comment by aleks-ndustrial** [comment]
+1

**Comment by yukha-dw** [comment]
+1

**Comment by dej4n** [comment]
+1

**Comment by joao-zanutto** [comment]
Recently, I came across a similar challenge: I needed multiple checkboxes in my `workflow_dispatch` event, and I needed to retrieve a list with the names of all the selected checkboxes.

To implement the multiple-choice checkboxes, you add as many single checkbox options as needed:
```yml
on:
  workflow_dispatch:
    inputs:
      option1:
        type: boolean
      option2:
        type: boolean
      option3:
        type: boolean
``` 

To retrieve the selected checkbox names, I created a custom action as I haven't found an easy way to do that without a for loop in bash ([GitHub Marketplace](https://github.com/marketplace/actions/get-selected)):

```yml
jobs:
  get-selected:
    name: Get Selected Checkboxes
    runs-on: ubuntu-latest
    outputs: # This needs to be set if you want to consume the output on another job
      selected: ${{ steps.get-selected-step.outputs.selected}}
    steps:
      - name: Checkout
        id: checkout
        uses: actions/checkout@v4

      - name: Get Selected
        id: get-selected-step
        uses: joao-zanutto/get-selected@v1

      - name: Consume on the same job
        run: echo ${{ steps.get-selected-step.outputs.selected }}

  consume-on-another-job:
    runs-on: ubuntu-latest
    name: Consume Output
    needs: get-selected
    steps:
      - name: Print
        run: echo ${{ needs.get-selected.outputs.selected }}
```

**Comment by aiskrenov** [comment]
+1

**Comment by valentinvs** [comment]
+1

**Comment by KostovaLia** [comment]
+1

**Comment by RapGeneral** [comment]
+1

**Comment by bangiev** [comment]
+1

**Comment by hourloop-ianlai** [comment]
+1

**Comment by rasikmhetre** [comment]
+1

**Comment by shayki5** [comment]
+1

**Comment by schweppes-0x** [comment]
Any updates ? How is this issue ongoing since 2021/2022 and still not implemented, what a shame :(

**Comment by pierre-sprk** [comment]
Tell what you want about jenkins. It had it.
How do you even go with a system that doesn't have such a basic feature?

**Comment by kailukowiak** [comment]
Chiming in here too. I'd really like this feature added.

**Comment by jake-ch** [comment]
+1

**Comment by orelviz** [comment]
+1

**Comment by MukundMohanan** [comment]
+1

**Comment by MaximilianoAdaro** [comment]
+1

**Comment by Meenu-Chandrasekaran** [comment]
+ 1

**Comment by mrygielskiWongaPl** [comment]
+1

**Comment by sorinirimies** [comment]
+1

**Comment by frederikstonge** [comment]
+1

**Comment by kmkvc3** [comment]
+1

**Comment by ThangZaThangWoodside** [comment]
👍 +1

**Comment by beccahallam** [comment]
+1 please this would be so helpful

**Comment by yourivisspringbok** [comment]
+1

**Comment by Steelsoul30** [comment]
+1

**Comment by pnv1** [comment]
+1

**Comment by carlydf** [comment]
+1

**Comment by stephenmathieson** [comment]
The flood of useless "+1" comments makes it very difficult to find any actual information here. Is this being worked on/on a road map somewhere?

**Comment by alxgomz** [comment]
True! I once pointed the actual feature request here https://github.com/actions/runner/issues/2076#issuecomment-1854462577
But that comment itself got hidden by the github comment "dup removal" feature so for now it has far less traction than this issue.. 🤷

**Comment by alxgomz** [comment]
> The flood of useless "+1" comments makes it very difficult to find any actual information here. Is this being worked on/on a road map somewhere?

True! I once pointed the actual feature request here https://github.com/actions/runner/issues/2076#issuecomment-1854462577
But that comment itself got hidden by the github comment "dup removal" feature so for now it has far less traction than this issue.. 🤷

**Comment by zachihazan** [comment]
+1

**Comment by joao-zanutto** [comment]
I'm adding this to bump my previous comment https://github.com/actions/runner/issues/2076#issuecomment-2198248106 where I introduced my own GitHub Action as a workaround for this problem as it might help someone.

The GitHub Action: https://github.com/marketplace/actions/get-selected

**Comment by Airsaid** [comment]
+1

**Comment by zenelbobi** [comment]
Is there anything else we can do besides upvotes?
It's been over two years and over 700 upvotes and still no update

**Comment by pnv1** [comment]
As a temporal solution I made [several bool inputs](https://github.com/ydb-platform/ydb/blob/main/.github/workflows/build_ydb_cli.yml#L6-L17) and then [formed a json matrix](https://github.com/ydb-platform/ydb/blob/main/.github/workflows/build_ydb_cli.yml#L42) of it for further jobs using a bunch of ifs and jq. May be will be useful for someone

**Comment by salvatoreboemia** [comment]
Please could you add this possibility to pass an array of teams from another workflow? 

thanks

**Comment by esultanbetclic** [comment]
+1

**Comment by korenlev** [comment]
+1

**Comment by ImJimmi** [comment]
+1

**Comment by kdurolek** [comment]
+1

**Comment by derekleegh** [comment]
+1

**Comment by vittoriopolverino** [comment]
+1

**Comment by Hagith** [comment]
That would be great improvement to our monorepo workflow with lerna independent versioning where we need to decide which packages should be graduated to stable :heart_hands:

**Comment by ShivaKumarKalvaTR** [comment]
+

**Comment by s-jorgenfroland** [comment]
+1

**Comment by cameronSMN** [comment]
+1

**Comment by hunterliu1003** [comment]
+1

**Comment by arikgreen** [comment]
+1

**Comment by HonzaV** [comment]
+1

**Comment by jlehoux-cvet** [comment]
+1

**Comment by rusowyler** [comment]
+1 👍

**Comment by nayanbabariya** [comment]
+1

**Comment by mllustosa** [comment]
+1

**Comment by tafaust** [comment]
+1

**Comment by sandstrom** [comment]
👍🏻

**Comment by nickgonzfoxeng** [comment]
+1

**Comment by SerhiiBasetskyi** [comment]
+1

**Comment by srinathreddydudi** [comment]
+1

**Comment by Oscaner** [comment]
+1

**Comment by ashishbhatt07** [comment]
+10

**Comment by AndyLysenko** [comment]
+1

**Comment by yx-altera** [comment]
+1

**Comment by qnp** [comment]
+1

**Comment by Himmelschrei** [comment]
+1

**Comment by martinsamouiller** [comment]
+1

**Comment by adsz01** [comment]
+1

**Comment by JoseNoguera-uala** [comment]
+1

**Comment by kxs581** [comment]
+1

**Comment by Andre-Medina** [comment]
+ 1

**Comment by marcelin** [comment]
Can you try this:

```

<img width="150" alt="Image" src="https://github.com/user-attachments/assets/0fa6c00e-8e16-4717-bcc6-2385e27fc67c" />

```

**Comment by marcelin** [comment]
Hi,

Can you try this?

```
name: "Multi-select"
run-name: 🎯${{ github.workflow}} by @${{ github.actor }}

on:
  workflow_dispatch:
    inputs:
      dev:
        description: 'Deploy to dev'
        default: false
        type: boolean
      staging:
        description: 'Deploy to staging'
        default: false
        type: boolean
      qat:
        description: 'Deploy to qat'
        default: false
        type: boolean
      prod:
        description: 'Deploy to prod'
        default: false
        type: boolean

jobs:
  get-selected:
    name: Get environment
    runs-on: ubuntu-latest
    outputs: 
      selectedJson: ${{ steps.get-selected-step.outputs.selected }}
    steps:
      - name: Checkout
        uses: actions/checkout@v4

      - name: Get Selected
        id: get-selected-step
        uses: joao-zanutto/get-selected@v1.1.1
        with:
          format: json

  plan-env:
    needs: get-selected
    runs-on: ubuntu-latest
    strategy:
      matrix:
        app: ${{ fromJSON(needs.get-selected.outputs.selectedJson) }}
    steps:
      - name: Checkout
        uses: actions/checkout@v4
     ....
```

<img width="150" alt="Image" src="https://github.com/user-attachments/assets/bd262a4c-7a03-4173-a312-5a9fa3cb6684" />

**Comment by dboriichuk** [comment]
> Hi,
> 
> Can you try this?
> 
> ```
> name: "Multi-select"
> run-name: 🎯${{ github.workflow}} by @${{ github.actor }}
> 
> on:
>   workflow_dispatch:
>     inputs:
>       dev:
>         description: 'Deploy to dev'
>         default: false
>         type: boolean
>       staging:
>         description: 'Deploy to staging'
>         default: false
>         type: boolean
>       qat:
>         description: 'Deploy to qat'
>         default: false
>         type: boolean
>       prod:
>         description: 'Deploy to prod'
>         default: false
>         type: boolean
> 
> jobs:
>   get-selected:
>     name: Get environment
>     runs-on: ubuntu-latest
>     outputs: 
>       selectedJson: ${{ steps.get-selected-step.outputs.selected }}
>     steps:
>       - name: Checkout
>         uses: actions/checkout@v4
> 
>       - name: Get Selected
>         id: get-selected-step
>         uses: joao-zanutto/get-selected@v1.1.1
>         with:
>           format: json
> 
>   plan-env:
>     needs: get-selected
>     runs-on: ubuntu-latest
>     strategy:
>       matrix:
>         app: ${{ fromJSON(needs.get-selected.outputs.selectedJson) }}
>     steps:
>       - name: Checkout
>         uses: actions/checkout@v4
>      ....
> ```
> 
> <img width="150" alt="Image" src="https://github.com/user-attachments/assets/bd262a4c-7a03-4173-a312-5a9fa3cb6684" />

Definitely, this works, the problem is that multi-choice is tightly coupled with limited number of inputs for workflow dispatch (10 max), so in many cases we don't want to have 8 boolean inputs, as it's necessary to have at least 5 another parameters

**Comment by origranot** [comment]
+100

**Comment by jordi-domingo** [comment]
+1

**Comment by marcelin** [comment]
+1

**Comment by GCbeheo** [comment]
+1

**Comment by Gnts** [comment]
+1

**Comment by alex-slytrunk** [comment]
+100

**Comment by potiuk** [comment]
+1

**Comment by benjyabr** [comment]
+1

**Comment by activtrak-jkelley** [comment]
+1 (on a 4 year old issue no doubt.... unbelievable!!!)

**Comment by bearjin** [comment]
+1

**Comment by danielnovello-pf** [comment]
+1

**Comment by kadolea** [comment]
+1 (require multi-select boolean inputs more than 10)

**Comment by Dumitru-Ene** [comment]
+1

**Comment by BosieIonut** [comment]
++1

**Comment by O1ahmad** [comment]
+1

**Comment by rgarcianavarro-facephi** [comment]
Some facts for our dear Github developers / product team / wathever 😠 

1. We're an organization paying for use GH and very sure that we're not the unique paying for its usage therefore we want continuos improvement not getting stuck in something such trivial like passing multiple values separated by comma to a CI job 😞 
2. We've here dozens and dozens of 👍🏻 (+1) which means that most likely the users wants and needs this feature
3. Almost 4 years later the only thing that I've seen throughout the different discussions and threads regarding this topic are just excuses from Github Team about UX and giving just tricky and ridiculous workarounds

What else you need dear Github to understand that we need this feature not because it looks fancy, we need it because it really solves a real world issue and that's why we are paying for your services ❓

**Comment by ccrawford4** [comment]
+1

**Comment by Ethonwu** [comment]
+1

**Comment by sousa-miguel** [comment]
+1

**Comment by alexivanov-ai** [comment]
I wonder when this is going to be added...... +1

**Comment by qnp** [comment]
+1

**Comment by craypineywoods** [comment]
last week... yesterday... 16 hours ago... 10 hours ago...

and .. now

+1

**Comment by joaoluiznaufel** [comment]
Looks like never, folks

**Comment by zach-cohen-milliman** [comment]
Folks, I can personally confirm that they're tracking this issue and many similar ones. Stop commenting +1.

**Comment by Benzou65** [comment]
What about +1 ?

**Comment by biskitsx** [comment]
+1

**Comment by inna-artiushenko** [comment]
+1

**Comment by thiagoescobar** [comment]
+1

**Comment by ravi1933** [comment]
+1

**Comment by flinn** [comment]
First... I mean...  Plus One, +1 , ➕1️⃣,  ✖️💯, 🤷

**Comment by mjshakespeare** [comment]
+1

**Comment by pulkit-pebble** [comment]
+1

**Comment by Alessandro-AP** [comment]
+1

**Comment by islampawapay** [comment]
+1

**Comment by danielwspie** [comment]
+1

**Comment by nbro10** [comment]
One of the richest companies in the world, Microsoft, who owns Github, cannot address this very basic need. Absolutely ridiculous and shameful.

**Comment by lukevo247** [comment]
+1

**Comment by judeokafor** [comment]
+1

**Comment by ivan-gaydamakin** [comment]
+1

**Comment by vaibhavmeena1** [comment]
+1

**Comment by gonisch** [comment]
+1

**Comment by Abdelrahman-incorta** [comment]
+1

**Comment by Terraformadillo** [comment]
Pretty sure github been waiting on my +1 this whole time so here it is.

+1

**Comment by bibek-shah-butterfly** [comment]
+1

**Comment by kruncher** [comment]
+1

**Comment by VatsalJagani** [comment]
+1

**Comment by kris-luminar** [comment]
+1

**Comment by dixonjoel** [comment]
+1

**Comment by mfroelund** [comment]
+1

**Comment by NorseGaud** [comment]
+1

**Comment by Ga13Ou** [comment]
+1

**Comment by BasixKOR** [comment]
> This issue is highly active. Reconsider commenting unless you have read all the comments and have something to add.

I'm unsure if this note is shown for everyone or just me but please stop commenting +1. It is not helpful for anyone nor it does not do anything other than spamming people's notifications. If you want to subscribe there's a subscribe button right next to your comment box.

**Comment by paul-uz** [comment]
> > This issue is highly active. Reconsider commenting unless you have read all the comments and have something to add.
> 
> I'm unsure if this note is shown for everyone or just me but please stop commenting +1. It is not helpful for anyone nor it does not do anything other than spamming people's notifications. If you want to subscribe there's a subscribe button right next to your comment box.

+1

**Comment by NorseGaud** [comment]
> > This issue is highly active. Reconsider commenting unless you have read all the comments and have something to add.
> 
> I'm unsure if this note is shown for everyone or just me but please stop commenting +1. It is not helpful for anyone nor it does not do anything other than spamming people's notifications. If you want to subscribe there's a subscribe button right next to your comment box.

It's not about that. It's about keeping the issue at the top of the list + making sure GIthub knows a lot of people actually want this. You can unsubscribe if you want.

**Comment by joaoluiznaufel** [comment]
+1, please act on this

**Comment by vpavic** [comment]
> It's about keeping the issue at the top of the list + making sure GIthub knows a lot of people actually want this.

I would assume folks at GitHub are able to figure that out without people spamming the issue.

<img width="1518" height="427" alt="Image" src="https://github.com/user-attachments/assets/72620306-8ade-4a70-9639-a19b97c9503a" />

**Comment by cdnarbona** [comment]
+1

**Comment by cristiancelering** [comment]
+1

**Comment by kolithakasun** [comment]
+1

**Comment by zenelbobi** [comment]
Maybe there is some hope after all

https://github.com/orgs/community/discussions/8774#discussioncomment-15039705

**Comment by mysoliev** [comment]
+1

**Comment by nebuk89** [comment]
👋 there is hope :) we are tracking this. It's not coming in the first quarter of next year (we will get our public roadmap tidied in the next week or so to show better expectations) but this is on the list of items we are tracking to improve manual workflows.

Keep giving 👍 on the top of the issue to show us where we should be looking and thank you for the patience

**Comment by dzhelnov** [comment]
Is there an update on feature placement in roadmap?

**Comment by nellai-m** [comment]
Adding my support for this!

@nebuk89 Since it's not on the public roadmap yet, is there any sense of when it might show up there?

Thanks!

**Comment by Swastikk27** [comment]
+1
